File: core/documents/semantic/semantic_diagram_engine.py
from __future__ import annotations

import re


def extract_semantic_diagrams(text: str):
    source = text or ""
    mermaid = re.findall(r"```mermaid\n(.*?)```", source, flags=re.DOTALL)
    plantuml = re.findall(r"```plantuml\n(.*?)```", source, flags=re.DOTALL)
    images = sorted(set(re.findall(r"!\[[^\]]*\]\(([^)]+)\)", source)))
    return {
        "mermaid": sorted({m.strip() for m in mermaid if m.strip()}),
        "plantuml": sorted({p.strip() for p in plantuml if p.strip()}),
        "image_refs": images,
    }

File: src/documents/semantic/semanticDiagramEngine.ts
/**
 * Converted from Python: core/documents/semantic/semantic_diagram_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../../runtime/pyCompat.js";

export function extractSemanticDiagrams(text: any): any {
  var source: any = py.or2(text, () => (""));
  var mermaid: any = py.reFindall("```mermaid\\n(.*?)```", source, "s");
  var plantuml: any = py.reFindall("```plantuml\\n(.*?)```", source, "s");
  var images: any = py.sorted(py.toSet(py.reFindall("!\\[[^\\]]*\\]\\(([^)]+)\\)", source, "")));
  return {"mermaid": py.sorted(py.toSet(py.iter(mermaid).filter((m: any) => py.truthy(py.strip(m))).map((m: any) => py.strip(m)))), "plantuml": py.sorted(py.toSet(py.iter(plantuml).filter((p: any) => py.truthy(py.strip(p))).map((p: any) => py.strip(p)))), "image_refs": images};
}
